scene-lib/src/annotations/io/IndexFileParser.java

```diff
@@ -579,19 +585,13 @@ private ScalarAFT parseScalarAFT() throws IOException, ParseException {
             if (abbreviate) {
               int i = name.lastIndexOf('.');
               if (i >= 0) {
-                String baseName = name.substring(i+1);
-                Set<String> set1 = scene.imports.get(name);
-                Set<String> set2 = scene.imports.get(baseName);
-                if (set1 == null) {
-                  set1 = new TreeSet<String>();
-                  scene.imports.put(name, set1);
-                }
-                if (set2 == null) {
-                  set2 = new TreeSet<String>();
-                  scene.imports.put(name, set2);
+                Set<String> importSet = scene.imports.get(annotationFullyQualifiedName);
+                if (importSet == null) {
+                  importSet = new TreeSet<String>();
+                  scene.imports.put(annotationFullyQualifiedName, importSet);
                 }
-                set1.add(name);
-                set2.add(name);
+                importSet.add(name);
```

Comment:
@mernst 's Comments in Original PR:
What happens if two annotations
```
    package1.MyAnno
    package2.MyAnno
```
were both inserted?

In that case, this would remove the existing mapping from `importSet` and create a new one, and the resulting file would be incorrect.

Therefore, I think this should check that `importSet` doesn't already contain name before calling `importSet.add(name)`.